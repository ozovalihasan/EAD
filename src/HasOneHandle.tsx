const HasOneHandle = ({visibility}: {visibility: string}) => (
  <svg className={`w-6 h-6 p-1 absolute -bottom-3 left-1/4 bg-slate-700 rounded-full -translate-x-1/2 ${visibility}`} viewBox="0 0 200 200">
    <path 
      d="m 90,0 v 200 h 20 V 0"
      className="fill-slate-50"
    />
  </svg>
)

export default HasOneHandle;

