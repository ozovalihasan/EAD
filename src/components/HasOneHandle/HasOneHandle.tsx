import useStore from 'zustandStore/store';
import { Handle, Position } from 'react-flow-renderer';

export const HasOneHandle = ({id} : {id: string}) => {
  const {
    onChangeAssociationType,
  } = useStore()

  return (
    <div className="w-6 h-6 absolute -bottom-3 left-1/4 bg-slate-700 rounded-full -translate-x-1/2" >
      <svg className="w-6 h-6" viewBox="0 0 200 200" >
        <path d="m 90,0 v 200 h 20 V 0" className="fill-slate-50" />
      </svg>

      <Handle
        onMouseDown={() => onChangeAssociationType("has_one", id)}
        className="opacity-50 justify-center items-center border-none w-6 h-6 bottom-0 left-1/2"
        type="source"
        position={Position.Bottom}
        id="bottom1"
      />

    </div>
  )
}


