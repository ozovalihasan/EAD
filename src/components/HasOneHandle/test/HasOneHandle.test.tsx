import { HasOneHandle } from '../HasOneHandle';
import { render, screen } from "@testing-library/react";

let renderReadyComponent: JSX.Element;

beforeEach(() => {
  renderReadyComponent = (
    <HasOneHandle nodeId="1" />
  );
});

describe('<HasOneHandle />', () => {
  it('calls another component with a prop with value "has_one"', () => {
    render(renderReadyComponent);
    expect(screen.getByText(/MockCustomHandle/i)).toBeInTheDocument();
    expect(screen.getByText(/has_one/i)).toBeInTheDocument();
  });


  it('renders correctly', () => {
    const renderedContainer = render(renderReadyComponent );
    expect(renderedContainer).toMatchSnapshot();
  });

});