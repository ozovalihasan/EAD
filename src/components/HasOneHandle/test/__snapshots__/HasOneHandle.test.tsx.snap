// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HasOneHandle /> renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-6 h-6 absolute left-1/4 -translate-x-1/2 bg-slate-700 rounded-full flex justify-center items-center"
      >
        <svg
          class="w-3 h-3 stroke-[40] stroke-slate-50"
          viewBox="0 0 200 200"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M 100,20 V 180"
          />
        </svg>
        nodeId: 
        1
        id: 
        bottom1
        handleType: 
        has_one
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-6 h-6 absolute left-1/4 -translate-x-1/2 bg-slate-700 rounded-full flex justify-center items-center"
    >
      <svg
        class="w-3 h-3 stroke-[40] stroke-slate-50"
        viewBox="0 0 200 200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M 100,20 V 180"
        />
      </svg>
      nodeId: 
      1
      id: 
      bottom1
      handleType: 
      has_one
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
